superagi/tools/knowledge_search/knowledge_search.py

```diff
@@ -49,7 +49,7 @@ def _execute(self, query: str):
         embedding_model = AgentExecutor.get_embedding(model_source, model_api_key)
         try:
             if vector_db_index.state == "Custom":
-                filters = {}
+                filters = None
```

Comment:
why do we need it?

Reply:
Sir this is for metadata filtering, for marketplace knowledge we have a default filter on knowledge name but for custom currently we are not giving user the option to specify filters